pkg/bgpv1/gobgp/server.go

```diff
@@ -265,7 +269,7 @@ func (g *GoBGPServer) getPeerConfig(ctx context.Context, n types.NeighborRequest
 	}
 
 	// Enable multi-hop for eBGP if non-zero TTL is provided
-	if g.asn != uint32(n.Neighbor.PeerASN) && *n.Neighbor.EBGPMultihopTTL > 1 {
+	if g.asn != uint32(n.Neighbor.PeerASN) && n.Neighbor.EBGPMultihopTTL != nil && *n.Neighbor.EBGPMultihopTTL > 1 {
```

Comment:
Hey @danehans, the commit talks of introducing a unit test, but doesn't mention why this change is necessary. Was this something the original code missed?

Reply:
@ldelossa thanks for the review. Yes, since `EBGPMultihopTTL` is a pointer, it should be checked for nil before dereferencing it. I caught this when adding one of the test cases in the PR. I will update the PR description accordingly.